base/set.jl

```diff
@@ -427,6 +427,34 @@ allunique(::Union{AbstractSet,AbstractDict}) = true
 
 allunique(r::AbstractRange) = !iszero(step(r)) || length(r) <= 1
 
+"""
+    allequal(itr) -> Bool
+
+Return `true` if all values from `itr` are equal when compared with [`isequal`](@ref).
+
+See also: [`unique`](@ref), [`allunique`](@ref), [`issorted`](@ref).
+
+# Examples
+```jldoctest
+julia> allequal([])
+true
+
+julia> allequal([1])
+true
+
+julia> allequal([1, 1])
+true
+
+julia> allequal([1, 2])
+false
+```
+"""
+allequal(itr) = isempty(itr) ? true : all(isequal(first(itr)), itr)
+
+allequal(c::Union{AbstractSet,AbstractDict}) = length(c) <= 1
```

Comment:
I find this surprising. My expectation was `allequal(Dict(1=>1, 2=>1)) == true`.

Reply:
`allequal`'s behavior should probably not discriminate on container's type and work the same for all iterators, and `Dict` iterates over key-value pairs.

Reply:
Could `allequal(Dict(1=>1, 2=>1))` be added to the doctest?

Reply:
Sure, I just added a similar example to the doctest.